pkg/operation/botanist/controlplane.go

```diff
@@ -400,25 +400,48 @@ func (b *Botanist) DeploySeedMonitoring() error {
 		values["alertmanager"].(map[string]interface{})["emailConfigs"] = emailConfigs
 	}
 
-	return b.ApplyChartSeed(filepath.Join(common.ChartPath, "seed-monitoring"), fmt.Sprintf("%s-monitoring", b.Shoot.SeedNamespace), b.Shoot.SeedNamespace, nil, values)
+	if err := b.ApplyChartSeed(filepath.Join(common.ChartPath, "seed-monitoring"), fmt.Sprintf("%s-monitoring", b.Shoot.SeedNamespace), b.Shoot.SeedNamespace, nil, values); err != nil {
+		return err
+	}
+
+	// If no mail receivers for the Shoot alerts provided than there is no need
+	// to run the Alertmanager and it can be removed from the control plane.
+	if len(emailConfigs) == 0 {
+		if err := b.deleteSeedAlertmanager(); err != nil && !apierrors.IsNotFound(err) {
+			return err
+		}
+	}
+	return nil
 }
 
 // DeleteSeedMonitoring will delete the monitoring stack from the Seed cluster to avoid phantom alerts
 // during the deletion process. More precisely, the Alertmanager and Prometheus StatefulSets will be
 // deleted.
 func (b *Botanist) DeleteSeedMonitoring() error {
```

Comment:
Any reason why the order was changed (first deleting prometheus, then alertmanager)? Previously, it was the other way round.